baggage/baggage.go

```diff
@@ -550,3 +530,124 @@ func (b Baggage) String() string {
 	}
 	return strings.Join(members, listDelimiter)
 }
+
+// parsePropertyInternal attempts to decode a Property from the passed string.
+func parsePropertyInternal(s string) (p Property, ok bool) {
+	// Attempting to parse the key.
+	index := skipSpace(s, 0)
+
+	keyStart := index
+	keyEnd := index
+	for _, c := range s[keyStart:] {
+		if !validateKeyChar(c) {
+			break
+		}
+		keyEnd++
+	}
+
+	if keyStart == keyEnd {
+		// Empty string after skipping whitespaces.
+		return
+	}
+
+	index = skipSpace(s, keyEnd)
+
+	p.key = s[keyStart:keyEnd]
```

Comment:
Can we set `p.key` only if a valid property is returned (when `ok == true`)?

Reply:
Of course I can, I didn't do it because it would just make the code more verbose, but will send the diff.